react-basic-practice/src/pages/DetailProductPage/index.jsx

```diff
@@ -75,6 +128,9 @@ const ProductDetail = () => {
             </>
           )
       }
+      {
+        openPopup && <Popup isSuccess={isSuccess} message={isSuccess ? 'The item added to your shopping bag' : 'Add to cart failed, please try again'} />
```

Comment:
This component should be able to be used globally.

Reply:
Do you mean use `useContext` for `openPopup` ?

Reply:
You need to put it somewhere so you won't have to define it again in other components.

Reply:
I will declare `openPopup` and `isSuccess` as global variables so it can be used in many places, however the message will change with each action and each component will have its own message so it will be declared separately.

Reply:
You should not hard code the message like this. Ideally messages should also be global state so you can send message anywhere you want.